packages/oats-runtime/src/runtime.ts

```diff
@@ -18,6 +22,23 @@ const tagKey = Symbol();
 
 type ScalarWithoutBrand<V> = V extends ShapedClass<infer S> ? ScalarWithoutBrand<S> : V;
 
+// helper to set necessary metadata  for valueclass instances
+export function instanceAssign(
```

Comment:
this will do the magic to set up value class instances up using Object.assign and the reflection types.